inst/tutorials/tidy-03-measurement1/tidy-03-measurement1.Rmd

```diff
@@ -0,0 +1,357 @@
+  group_by(classtype) %>%
+  count() # of uniques
+
+## make a barplot with ggplot
+classcounts %>% ggplot(aes(x = classtype, y = n)) +
+  geom_bar(stat='identity') 
+
+```
+
+```{r barplot-check}
+grade_code("Awesome. The graph is looking a little unhelpful, though. Let's spruce it up.")
+```
+
+### Making the barplot readable
+
+The default barplot usually isn't all that readable.
+
+#### Exercises
+
+- Use the function `scale_x_discrete(labels = c())` to name the categories of the dependent variable. Use `"Small class"` for 1, `"Regular class"` for 2, and `"Regular class with aid"` for 3. Remember to factorize the variable `classtype` in `aes()` with `factor()`, otherwise the x values will be defined as numeric rather than discrete. 
+- You are encouraged to use pipes.
+
+```{r barplot-labels, exercise = TRUE}
+## Use the function scale_x_discrete(labels = c()) 
+## i.e., scale_x_discrete(labels = c(
+##     "1" = "Small class", 
+##     "2" = "" ,
+##     ...))
+
+
+
+```
+
+```{r barplot-labels-solution}
+classcounts %>% ggplot(aes(x=factor(classtype), y = n)) +
+  geom_bar(stat='identity') + 
+   scale_x_discrete(labels = c(
+     "1" = "Small class", 
+     "2" = "Regular class", 
+     "3" = "Regular class with aid"))
+
+```
+
+
+- Use the function `labs` to add the lables and title to the plot. Lable the x axis `"Classnames"`, the y axis `"Number of students"`, and the title `"The Distribution of Students in Different Class Types"`.
+- You are encouraged to use pipes.
+
+
+```{r barplot-lab, exercise = TRUE}
+## Use the function `labs` and `title`
+
+
+
+```
+
+```{r barplot-lab-solution}
+classcounts %>% ggplot(aes(x=factor(classtype), y = n)) +
+  geom_bar(stat='identity') + 
+   scale_x_discrete(labels = c(
+     "1" = "Small class", 
+     "2" = "Regular class", 
+     "3" = "Regular class with aid")) +
+  labs( x = "Classnames", 
+        y = "Number of students", 
+        title = "The Distribution of Students in Different Class Types")
+```
+
+
+### Histograms
+
+For quantitative (numerical) variables, the barplot won't work because there are too many unique values. In this case, you will often use a histogram to visualize the a numerical variable.
+
+#### Exercises
+- With `ggplot()` function, use the `geom_histogram()` geom to create a histogram for the `g4math` variable in the `star` data frame.
+- You are encouraged to use pipes.
+
+```{r hist, exercise = TRUE}
+## create a histogram of g4math with ggplot
+
+```
+
+
+```{r hist-solution}
+## create a histogram of g4math with ggplot
+star %>% ggplot(aes(x = g4math)) +
+  geom_histogram()
+```
+
+```{r hist-check}
+grade_code("Great job, though the graph is a bit spartan. Let's make it more readable.")
+```
+
+### Sprucing up the histogram
+
+Same as the barplot, there are several arguments you can pass to the `ggplot()` function that will improve its readability:
+
+- `aes(y = ..density..)` argument in `geom_histogram` allows you to make a density plot with `ggplot`.
```

Comment:
Think the ".." notation is superseded, and we should use `stat(density)` (or `after_stat(density)` for even more up to date) instead.

Also `y = stat(density*width)` instead of `y = stat(density)` will give the proportion of each bar, instead of the density.

Reply:
Great comments; I have replaced the legacy code with `after_stat(density)` and added an extra exercise on plotting the proportion of the bins.